File: rzops-web/src/lib/types/role.ts
export interface RoleResponse {
  id: string;
  code: string;
  name: string;
  description: string | null;
  is_builtin: boolean;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

export interface RoleListResponse {
  data: RoleResponse[];
  count: number;
}

export interface RoleDetailResponse {
  role: RoleResponse;
  permissions: string[];
}

export interface CreateRoleRequest {
  code: string;
  name: string;
  description?: string | null;
  permissions: string[];
}

export interface UpdateRoleRequest {
  name?: string;
  description?: string | null;
  is_active?: boolean;
  permissions?: string[];
}

/** 权限点分组定义：资源 → 可勾选的操作 */
export interface PermissionResource {
  key: string;
  label: string;
  /** 是否系统级（单权限点 system:*） */
  system?: boolean;
}

File: rzops-api/domain/src/models/role.rs
use chrono::{DateTime, Utc};
use uuid::Uuid;

/// 角色 — maps to role 表。
/// 预置角色（admin/ops/audit/viewer）is_builtin=true，不可删除；
/// 自定义角色由管理员创建并勾选权限点。
#[derive(Debug, Clone)]
pub struct Role {
    pub id: Uuid,
    pub code: String,
    pub name: String,
    pub description: Option<String>,
    pub is_builtin: bool,
    pub is_active: bool,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
    pub deleted_at: Option<DateTime<Utc>>,
}
